### ISSUE TITLE
Required Fields for @rjsf/antd

### ISSUE THREAD

**Issue by georgiossalon** [rjsf-team/react-jsonschema-form#3743]
### Prerequisites

- [X] I have searched the existing issues
- [X] I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [X] I have read the [documentation](https://rjsf-team.github.io/react-jsonschema-form/docs)
- [X] Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net), [Codesandbox.io](https://codesandbox.io) or preferably a [shared playground link](https://rjsf-team.github.io/react-jsonschema-form/) demonstrating the issue.

### What theme are you using?

antd

### Version

5.8.2

### Current Behavior

Input fields that are set as required by the scheme are missing the required property in the browser.

In the following code, if you switch to the MaterialUI or the default Form component you do get the expected result. When using the antd Form component via the withTheme you don't.

Here is a dummy Form:

```
import React, {createRef} from "react";
import "./App.less";
import {withTheme} from "@rjsf/core";
import Form from '@rjsf/material-ui';
import validator from "@rjsf/validator-ajv8";

import {Theme as AntDTheme} from '@rjsf/antd';
import {Popconfirm} from "antd";

const RJSFForm = withTheme(AntDTheme);

const uiSchema = {
    "ui:disabled": false,
    "ui:order": [
        "Name1",
        "Number 1",
    ],
    Name1: {
        "ui:disabled": false
    },
    "Number 1": {
        "ui:disabled": false
    },
};

const schema = {
    type: "object",
    $schema: "http://json-schema.org/draft-07/schema#",
    required: ["Name1"],
    "ui:order": [
        "Name1",
        "Number 1",
    ],
    properties: {
        "Name1": {
            type: "string"
        },
        "Number 1": {
            type: "number",
            default: 0
        },
    },
    $id: "http://json-schema.org/draft-07/schema#"
};

function App() {
    const formRef = createRef();
    const submitRef = createRef();
    const onError = (errors) => console.log(errors);

    return (
        <div className="App">
            <RJSFForm
                schema={schema}
                uiSchema={uiSchema}
                validator={validator}
                onError={onError}
                ref={formRef}
                formData={{
                    "Number 1": 0,
                }}
            >
                <button hidden={true} ref={submitRef}/>
            </RJSFForm>
            <button
                onClick={() => {
                    submitRef.current.click();
                }}
            >
                submit
            </button>

        </div>
    );
}


export default App
```

### Expected Behavior

In the browser, the input tags should get the required property when the set into the schema.

### Steps To Reproduce

Sandbox: [CodeSandbox](https://codesandbox.io/s/suspicious-http-v34hgn?file=/src/App.js)

### Environment

```markdown
- OS: -
- Node: v16
- npm: v8
```

**Comment by nickgros** [comment]
[Here's a playground link](https://rjsf-team.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6eyJOdW1iZXIgMSI6MH0sInNjaGVtYSI6eyJ0eXBlIjoib2JqZWN0IiwiJHNjaGVtYSI6Imh0dHA6Ly9qc29uLXNjaGVtYS5vcmcvZHJhZnQtMDcvc2NoZW1hIyIsInJlcXVpcmVkIjpbIk5hbWUxIl0sInByb3BlcnRpZXMiOnsiTmFtZTEiOnsidHlwZSI6InN0cmluZyJ9LCJOdW1iZXIgMSI6eyJ0eXBlIjoibnVtYmVyIiwiZGVmYXVsdCI6MH19fSwidWlTY2hlbWEiOnsidWk6ZGlzYWJsZWQiOmZhbHNlLCJ1aTpvcmRlciI6WyJOYW1lMSIsIk51bWJlciAxIl0sIk5hbWUxIjp7InVpOmRpc2FibGVkIjpmYWxzZX0sIk51bWJlciAxIjp7InVpOmRpc2FibGVkIjpmYWxzZX19LCJ0aGVtZSI6ImFudGQiLCJsaXZlU2V0dGluZ3MiOnsic2hvd0Vycm9yTGlzdCI6InRvcCIsInZhbGlkYXRlIjpmYWxzZSwiZGlzYWJsZWQiOmZhbHNlLCJub0h0bWw1VmFsaWRhdGUiOmZhbHNlLCJyZWFkb25seSI6ZmFsc2UsIm9taXRFeHRyYURhdGEiOmZhbHNlLCJsaXZlT21pdCI6ZmFsc2UsImV4cGVyaW1lbnRhbF9kZWZhdWx0Rm9ybVN0YXRlQmVoYXZpb3IiOnsiYXJyYXlNaW5JdGVtcyI6InBvcHVsYXRlIiwiZW1wdHlPYmplY3RGaWVsZHMiOiJwb3B1bGF0ZUFsbERlZmF1bHRzIn19fQ==). The antd UI indicates that the field is required, and Ajv validation works as expected. The main difference that I see is that the antd theme is missing HTML5 validation.

**Comment by georgiossalon** [comment]
@nickgros Yeah, the field looks like it would be a required one but in the input tag, the required property is missing. The same applies to your playground link.
